File: backend/app/Models/Tenant.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;

class Tenant extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'name',
        'subdomain',
        'business_type',
        'size_category',
        'national_id',
        'economic_code',
        'phone',
        'address',
        'settings',
        'is_active',
        'trial_ends_at',
    ];

    protected $casts = [
        'settings' => 'array',
        'is_active' => 'boolean',
        'trial_ends_at' => 'datetime',
    ];

    public function users(): HasMany
    {
        return $this->hasMany(User::class);
    }

    public function theme(): HasOne
    {
        return $this->hasOne(TenantTheme::class);
    }

    public function roles(): HasMany
    {
        return $this->hasMany(Role::class);
    }

    public function warehouses(): HasMany
    {
        return $this->hasMany(Warehouse::class);
    }

    public function products(): HasMany
    {
        return $this->hasMany(Product::class);
    }

    public function parties(): HasMany
    {
        return $this->hasMany(Party::class);
    }

    public function accounts(): HasMany
    {
        return $this->hasMany(Account::class);
    }

    public function fiscalYears(): HasMany
    {
        return $this->hasMany(FiscalYear::class);
    }

    public function isOnTrial(): bool
    {
        return $this->trial_ends_at && $this->trial_ends_at->isFuture();
    }
}

File: packages/shared_ui/lib/src/models/tenant.dart
class Tenant {
  final int id;
  final String name;
  final String subdomain;
  final String? businessType;
  final String? sizeCategory;

  Tenant({
    required this.id,
    required this.name,
    required this.subdomain,
    this.businessType,
    this.sizeCategory,
  });

  factory Tenant.fromJson(Map<String, dynamic> json) => Tenant(
        id: json['id'] as int,
        name: json['name'] as String,
        subdomain: json['subdomain'] as String,
        businessType: json['business_type'] as String?,
        sizeCategory: json['size_category'] as String?,
      );
}
